framework/src/actions/CreateDisplacedProblemAction.C

```diff
@@ -26,23 +26,35 @@ InputParameters validParams<InitDisplacedProblemAction>()
   return params;
 }
 
-InitDisplacedProblemAction::InitDisplacedProblemAction(InputParameters parameters) :
+CreateDisplacedProblemAction::CreateDisplacedProblemAction(InputParameters parameters) :
     Action(parameters)
 {
 }
 
-InitDisplacedProblemAction::~InitDisplacedProblemAction()
+CreateDisplacedProblemAction::~CreateDisplacedProblemAction()
 {
 }
 
 void
-InitDisplacedProblemAction::act()
+CreateDisplacedProblemAction::act()
 {
-  if (isParamValid("displacements"))
+  if (isParamValid("displacements") && _displaced_mesh.get() != NULL)
```

Comment:
I am on it, thanks.

On Fri, Aug 14, 2015 at 4:16 PM, Derek Gaston <notifications@github.com>
wrote:

> In framework/src/actions/CreateDisplacedProblemAction.C
> <https://github.com/idaholab/moose/pull/5564#discussion_r37124657>:
>
> >  {
> > -  if (isParamValid("displacements"))
> > +  if (isParamValid("displacements") && _displaced_mesh.get() != NULL)
>
> I think it may be better to just:
>
> if (_displaced_mesh.get() == NULL)
>   mooseError("displacements were set but a displaced mesh wasn't created!")
>
> Inside the if here...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5564/files#r37124657>.
>